### DISCUSSION TITLE
Recommended way of using Mobx with React in 2022?

### DISCUSSION THREAD

**Discussion by benjick** [mobxjs/mobx#3253] [Q&A] [answered]
Hello!

Been a long time since I was using MST and I want to come back, but looking at some community examples it looks kinda tedious when you're used to the standard hooks and zustand. Do I really need a Provider + hook AND the observer HOC? The React integration page doesn't even mention hooks. I would want to just be able to use a hook in the components I need the data. Is that possible?

Thanks in advance!

**Comment by urugator** [comment, marked as answer]
You have to wrap your state in `observable` and your components in `observer`. In exchange, you don't have to worry about immutability, memoization, selectors, reducers, subscriptions.

> Provider

It's up to you how you get state into your component - react context/import/local state.

> Is that possible?

No, because we have to wrap over rendering logic. Also it's no more tedious than using `React.memo`, which you would have to use anyway to achieve similar perf benefits.

**Reply by benjick** [reply]
I understand, thanks for the reply!

**Reply by BrianHung** [reply]
What's the way to use memo instead of observer? I'd prefer minimal mobx boilerplate and rely on default react methods for the view if possible.

Is there not a way to move the observer functionality into the hook?
